clang/lib/AST/ByteCode/InterpStack.cpp

```diff
@@ -26,33 +26,33 @@ InterpStack::~InterpStack() {
     std::free(Chunk);
   Chunk = nullptr;
   StackSize = 0;
-#ifndef NDEBUG
   ItemTypes.clear();
-#endif
 }
 
 // We keep the last chunk around to reuse.
 void InterpStack::clear() {
-  if (!Chunk)
-    return;
-
-  if (Chunk->Next)
-    std::free(Chunk->Next);
-
-  assert(Chunk);
-  StackSize = 0;
-#ifndef NDEBUG
-  ItemTypes.clear();
-#endif
+  for (PrimType Item : llvm::reverse(ItemTypes)) {
+    TYPE_SWITCH(Item, { this->discard<T>(); });
```

Comment:
Seeing `TYPE_SWITCH` in action, I don't know if the name really reflects what it is doing. Your not really doing something different based on the type but apply "templated code" uniformly across all the types. Not a comment for this PR and I am having trouble coming up w/ a better name.